net/src/client/http/mod.rs

```diff
@@ -0,0 +1,62 @@
+use std::str::FromStr;
+use std::sync::Arc;
+
+use futures03::future::BoxFuture;
+use surf::http::{self, Response, Uri};
+use surf::middleware::{Body, HttpClient as SurfHttpClient};
+use surf::Client;
+
+type Request = http::Request<Body>;
+
+/// A surf-alike HTTP client that additionally supports proxies (HTTP(S), SOCKS4 and SOCKS5)
+#[derive(Clone, Debug)]
+pub struct WitnetHttpClient {
+    client: Arc<isahc::HttpClient>,
+}
+
+impl WitnetHttpClient {
+    /// Create a new `WitnetHttpClient`
+    pub fn new(proxy: &Option<String>) -> Self {
```

Comment:
```suggestion
    pub fn new(proxy: Option<&str>) -> Self {
```

And on the caller side, you can use `.as_deref()` to convert `Option<String>` into `Option<&str>`.